### DISCUSSION TITLE
Setting up VLAN before fetching ignition config

### DISCUSSION THREAD

**Discussion by 0xalex88** [flatcar/Flatcar#1441] [Q&A] [answered]
I was looking at [this](https://www.flatcar.org/docs/latest/provisioning/ignition/network-configuration/#using-static-ip-addresses-with-ignition) docs page to be able to set vlan tagging and static IP before trying to get the ignition config.
I'm using matchbox to return this IPXE script (ipxe vlan works fine by using `vcreate --tag 1 net1`):

```
#!ipxe
kernel /assets/flatcar/3815.2.2/flatcar_production_pxe.vmlinuz initrd=flatcar_production_pxe_image.cpio.gz flatcar.config.url=http://10.1.0.1:8080/ignition?uuid=${uuid}&mac=${mac:hexhyp} flatcar.first_boot=yes ip=10.1.2.1 netmask=255.255.0.0 ksdevice=vlan.1 vlan=vlan.1:enp4s0f1
initrd /assets/flatcar/3815.2.2/flatcar_production_pxe_image.cpio.gz
boot
```

however when booting the kernel rejects the arguments with:
```
Unknown kernel command line parameters "ip=10.1.2.1 netmask=255.255.0.0 ksdevice=vlan.1 vlan=vlan.1:enp4s0f1", will be passed to user space
```

Any idea on how to get the private instance set up?

Update:

I've also tried with `vlan=enp4s0f1.1:enp4s0f1 ip=10.1.2.1:::255.255.0.0::enp4s0f1.1:::::` and I'm getting

```
99-parse-ip-for-networkd.sh[497]: Warning: Empty autoconf values default to dhcp
dracut: FATAL: For argument 'ip=10.1.2.1:::255.255.0.0::enp4s0f1.1:::::'\nSorry, setting client-ip does not  make sense for dhcp
```

Update 2:

Fixed the error above by adding `none` but still no vlan, maybe dracut used by flatcar doesn't support adding the vlan config?

**Comment by jepio** [comment, marked as answer]
This is the script responsible for setting up networking from the initrd:
https://github.com/flatcar/bootengine/blob/8da532c809c89a9c434ada0fa9532a1c1bf49f4c/dracut/03flatcar-network/parse-ip-for-networkd.sh

reading it, it doesn't seem to support setting up vlan

**Reply by 0xalex88** [reply]
Thanks, I guess there's no way then
